### DISCUSSION TITLE
QPInverseProblemSolver Question: Strange Result after adding Interative collision

### DISCUSSION THREAD

**Discussion by TeoREN0217** [SofaDefrost/SoftRobots#180] [Help - Programming with SoftRobots]
Hi! I've been working with a trunk-like soft robot recently and I found `QPInverseProblemSolver` has very strange results. Here is the problem:
Without collision in the environment, the tip has a normal result like:
![fc5060e25b94b8d3b28ca8937f0e68c](https://user-images.githubusercontent.com/60545892/181878955-b3e162d7-436a-48d4-b20c-04d8c8eb9604.png)![afea3dd8e5da3fed808d4199afd7939](https://user-images.githubusercontent.com/60545892/181878970-5a05e3cf-7c9a-444b-8646-c4d66faffb4d.png)
But after adding interactive collision, the result always makes the robot tip lean to the left (in different youngmodules, positions, robot FEM mesh after I trying to find the problem but always left!) like:
![cfe8c127562d2bb97f18420fffec311](https://user-images.githubusercontent.com/60545892/181879093-538b2646-0e10-4635-8364-f07f82fe9331.png)![微信图片_20220730150641](https://user-images.githubusercontent.com/60545892/181879164-0c975528-e3d8-4079-8c5e-0a9546905224.png)![f2e34b2d30e755d30268b89aa2e04f2](https://user-images.githubusercontent.com/60545892/181879382-cb297114-f531-4beb-8d3f-508256f8f1c0.jpg)

I'm pretty sure it's not the environment collision model because I tried to cut different parts from the model and it didn't fix the problem. Also, there is a big force from on of the cable actuator makes the robot transform like：
![e7219ae584516a8c52330cfa5707e4f](https://user-images.githubusercontent.com/60545892/181879490-131fd4b7-12cd-4629-b1d7-3d007ebbe2ae.png)
I tried to add a constraint of X axis but it still make robot transform. The cable position is symmetrical and the No.1 cable actuator has bigger force than No.2 cable actuator because after animating a while, No.2 cable force will be 0 and only No.1 has force and that why the robot lean to left. These whole thing worked in Inverse mode so the `QPInverseProblemSolver` solved the value. I was wondering is `QPInverseProblemSolver` has some bugs or something wrong with my model?
Here is the displacement and force exported. For no matter the robot and it's actuator or the environment collision models are all symmetrical, I think the two cable should has a similar result. 
![208bdf5a4670cf73a65280e2028fc21](https://user-images.githubusercontent.com/60545892/181880392-e427cb80-1d51-4d78-9139-afea5f240815.jpg)
![2998938e76a77e301c707be5ea11a87](https://user-images.githubusercontent.com/60545892/181880476-44c1dff5-2475-42ea-bd3f-309b3d35a533.jpg)
Higher force make the displacement lower and make the robot transformed.

**Comment by TeoREN0217** [comment]
Hi~ @hugtalbot @EulalieCoevoet
Is there any other solver that can replace `QPInverseProblemSolver`?

**Comment by EulalieCoevoet** [comment]
Hi @TeoREN0217, is it possible to test your scene?
It is hard to understand what could be wrong with the few info you gave. I need to see how you've defined your cables (passing points, attached points, limits), and I also need to see the contact areas.

**Reply by TeoREN0217** [reply]
Hi @EulalieCoevoet, Thanks for your reply! Actually I have found the problem and I think it caused by cable and mesh model are huddled together and I tried to split them and it was fixed! Thanks again ! I have another question about `SlidingConstraint` and I will start another question.
